### PR TITLE
dpdk: count number of packets with invalid checksum v1

### PR DESCRIPTION
**Pull request by lukashino**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [X] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/5553) ticket:

Describe changes:
- fix a lost mbuf case when Suricata packet allocation failed, Suricata continued but the mbuf was never freed
- support counting packets with an invalid checksum
- add XSTATS output for a more detailed debug of NIC stats

### CODE REVIEW

File: src/decode.h [hunk 1]
```diff
@@ -41,13 +41,13 @@
 #include "util-napatech.h"
 #endif /* HAVE_NAPATECH */
 
-
```
Comment (lukashino) [inline src/decode.h:44]: Clang-format did it, so I rather went with it. :man_shrugging:

File: src/source-dpdk.c [hunk 2]
```diff
@@ -491,6 +509,51 @@ static TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void
...
+    struct rte_eth_xstat *xstats;
+    struct rte_eth_xstat_name *xstats_names;
+    static const char *stats_border = "_______";
+
+    len = rte_eth_xstats_get(port_id, NULL, 0);
+    if (len < 0)
+        FatalError(SC_ERR_DPDK_CONF, "Error (%s) getting count of rte_eth_xstats failed on port %s",
+                rte_strerror(-len), port_name);
+
+    xstats = calloc(len, sizeof(*xstats));
```
Comment (victorjulien) [inline src/source-dpdk.c:524]: SCCalloc

File: src/source-dpdk.c [hunk 3]
```diff
@@ -491,6 +509,51 @@ static TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void
...
+    if (xstats == NULL)
+        FatalError(SC_ERR_MEM_ALLOC, "Failed to allocate memory for the rte_eth_xstat structure");
+
+    ret = rte_eth_xstats_get(port_id, xstats, len);
+    if (ret < 0 || ret > len) {
+        free(xstats);
+        FatalError(SC_ERR_DPDK_CONF, "Error (%s) getting rte_eth_xstats failed on port %s",
+                rte_strerror(-ret), port_name);
+    }
+    xstats_names = calloc(len, sizeof(*xstats_names));
```
Comment (victorjulien) [inline src/source-dpdk.c:534]: SCCalloc

File: src/source-dpdk.c [hunk 4]
```diff
@@ -491,6 +509,51 @@ static TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void
...
+    if (xstats_names == NULL) {
+        free(xstats);
```
Comment (victorjulien) [inline src/source-dpdk.c:536]: SCFree

File: src/source-dpdk.c [hunk 5]
```diff
@@ -491,6 +509,51 @@ static TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void
     SCReturnInt(TM_ECODE_FAILED);
 }
 
+static void PrintDPDKPortXstats(uint32_t port_id, char port_name[RTE_ETH_NAME_MAX_LEN])
```
Comment (victorjulien) [inline src/source-dpdk.c:512]: why not use `const char *port_name` here?

File: src/source-dpdk.c [hunk 6]
```diff
@@ -491,6 +509,51 @@ static TmEcode ReceiveDPDKThreadInit(ThreadVars *tv, const void *initdata, void
...
+        FatalError(SC_ERR_DPDK_CONF, "Error (%s) getting names of rte_eth_xstats failed on port %s",
+                rte_strerror(-ret), port_name);
+    }
+    for (i = 0; i < len; i++) {
+        if (xstats[i].value > 0)
+            SCLogPerf("Port %u (%s): %s %s:\t\t%" PRIu64, port_id, port_name, stats_border,
+                    xstats_names[i].name, xstats[i].value);
+    }
+
+    fflush(stdout);
```
Comment (victorjulien) [inline src/source-dpdk.c:552]: Not needed when using our `SCLog*` calls.

**Review by victorjulien — CHANGES_REQUESTED** [review]
Can you add example output? Of the `PrintDPDKPortXstats` function?

**Comment by lukashino** [comment]
> Can you add example output? Of the `PrintDPDKPortXstats` function?

It prints out non-zero counters so there might be more or less output depending on how busy counters were. 
Here is an example:

```
[3390] 22/9/2022 -- 08:44:34 - (tm-threads.c:1927) <Notice> (TmThreadWaitOnThreadInit) -- Threads created -> W: 1 FM: 1 FR: 1   Engine started.                                                                                                       
[3390] 22/9/2022 -- 08:45:13 - (suricata.c:2776) <Notice> (SuricataMainLoop) -- Signal Received.  Stopping engine.
[3390] 22/9/2022 -- 08:45:14 - (suricata.c:1168) <Info> (SCPrintElapsedTime) -- time elapsed 42.095s
[3401] 22/9/2022 -- 08:45:15 - (flow-manager.c:1124) <Perf> (FlowRecycler) -- 41758 flows processed
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_packets:              999999
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_bytes:                760030582
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_packets:                999999
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_bytes:          760030582
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_packets:           963516
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_bytes:             724508317
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_packets:         36483
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_bytes:           39522261
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_packets:               999999
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_bytes:         764030578
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:587) <Perf> (ReceiveDPDKThreadExitStats) -- Total RX stats of 0000:3b:00.1: packets 999999 bytes: 760030582 missed: 0 errors: 0 nombufs: 0
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:598) <Perf> (ReceiveDPDKThreadExitStats) -- (W#01-0000..00.1) received packets 999999
[3390] 22/9/2022 -- 08:45:16 - (counters.c:871) <Info> (StatsLogSummary) -- Alerts: 0
[3390] 22/9/2022 -- 08:45:16 - (ippair.c:296) <Perf> (IPPairPrintStats) -- ippair memory usage: 414144 bytes, maximum: 16777216
[3390] 22/9/2022 -- 08:45:17 - (host.c:299) <Perf> (HostPrintStats) -- host memory usage: 398144 bytes, maximum: 335544320
[3390] 22/9/2022 -- 08:45:17 - (detect-engine-build.c:1784) <Info> (SigAddressCleanupStage1) -- cleaning up signature grouping structure... complete
[3390] 22/9/2022 -- 08:45:17 - (util-device.c:355) <Notice> (LiveDeviceListClean) -- Stats for '0000:3b:00.1':  pkts: 999999, drop: 0 (0.00%), invalid chksum: 0
[3390] 22/9/2022 -- 08:45:17 - (util-dpdk.c:52) <Info> (DPDKCloseDevice) -- Closing device 0000:3b:00.1
```

**Comment by lukashino** [comment]
Continues in #7900

**Comment by victorjulien** [comment]
> > Can you add example output? Of the `PrintDPDKPortXstats` function?
> 
> It prints out non-zero counters so there might be more or less output depending on how busy counters were. Here is an example:
> 
> ```
> [3390] 22/9/2022 -- 08:44:34 - (tm-threads.c:1927) <Notice> (TmThreadWaitOnThreadInit) -- Threads created -> W: 1 FM: 1 FR: 1   Engine started.                                                                                                       
> [3390] 22/9/2022 -- 08:45:13 - (suricata.c:2776) <Notice> (SuricataMainLoop) -- Signal Received.  Stopping engine.
> [3390] 22/9/2022 -- 08:45:14 - (suricata.c:1168) <Info> (SCPrintElapsedTime) -- time elapsed 42.095s
> [3401] 22/9/2022 -- 08:45:15 - (flow-manager.c:1124) <Perf> (FlowRecycler) -- 41758 flows processed
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_packets:              999999
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_bytes:                760030582
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_packets:                999999
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_bytes:          760030582
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_packets:           963516
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_bytes:             724508317
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_packets:         36483
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_bytes:           39522261
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_packets:               999999
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_bytes:         764030578
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:587) <Perf> (ReceiveDPDKThreadExitStats) -- Total RX stats of 0000:3b:00.1: packets 999999 bytes: 760030582 missed: 0 errors: 0 nombufs: 0
> [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:598) <Perf> (ReceiveDPDKThreadExitStats) -- (W#01-0000..00.1) received packets 999999
> [3390] 22/9/2022 -- 08:45:16 - (counters.c:871) <Info> (StatsLogSummary) -- Alerts: 0
> [3390] 22/9/2022 -- 08:45:16 - (ippair.c:296) <Perf> (IPPairPrintStats) -- ippair memory usage: 414144 bytes, maximum: 16777216
> [3390] 22/9/2022 -- 08:45:17 - (host.c:299) <Perf> (HostPrintStats) -- host memory usage: 398144 bytes, maximum: 335544320
> [3390] 22/9/2022 -- 08:45:17 - (detect-engine-build.c:1784) <Info> (SigAddressCleanupStage1) -- cleaning up signature grouping structure... complete
> [3390] 22/9/2022 -- 08:45:17 - (util-device.c:355) <Notice> (LiveDeviceListClean) -- Stats for '0000:3b:00.1':  pkts: 999999, drop: 0 (0.00%), invalid chksum: 0
> [3390] 22/9/2022 -- 08:45:17 - (util-dpdk.c:52) <Info> (DPDKCloseDevice) -- Closing device 0000:3b:00.1
> ```

That looks a bit... custom. Why the `___`s? Why the spaces/tabs between name and values?

**Comment by lukashino** [comment]
> > > Can you add example output? Of the `PrintDPDKPortXstats` function?
> > 
> > 
> > It prints out non-zero counters so there might be more or less output depending on how busy counters were. Here is an example:
> > ```
> > [3390] 22/9/2022 -- 08:44:34 - (tm-threads.c:1927) <Notice> (TmThreadWaitOnThreadInit) -- Threads created -> W: 1 FM: 1 FR: 1   Engine started.                                                                                                       
> > [3390] 22/9/2022 -- 08:45:13 - (suricata.c:2776) <Notice> (SuricataMainLoop) -- Signal Received.  Stopping engine.
> > [3390] 22/9/2022 -- 08:45:14 - (suricata.c:1168) <Info> (SCPrintElapsedTime) -- time elapsed 42.095s
> > [3401] 22/9/2022 -- 08:45:15 - (flow-manager.c:1124) <Perf> (FlowRecycler) -- 41758 flows processed
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_packets:              999999
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_bytes:                760030582
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_packets:                999999
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_bytes:          760030582
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_packets:           963516
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_bytes:             724508317
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_packets:         36483
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_bytes:           39522261
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_packets:               999999
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_bytes:         764030578
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:587) <Perf> (ReceiveDPDKThreadExitStats) -- Total RX stats of 0000:3b:00.1: packets 999999 bytes: 760030582 missed: 0 errors: 0 nombufs: 0
> > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:598) <Perf> (ReceiveDPDKThreadExitStats) -- (W#01-0000..00.1) received packets 999999
> > [3390] 22/9/2022 -- 08:45:16 - (counters.c:871) <Info> (StatsLogSummary) -- Alerts: 0
> > [3390] 22/9/2022 -- 08:45:16 - (ippair.c:296) <Perf> (IPPairPrintStats) -- ippair memory usage: 414144 bytes, maximum: 16777216
> > [3390] 22/9/2022 -- 08:45:17 - (host.c:299) <Perf> (HostPrintStats) -- host memory usage: 398144 bytes, maximum: 335544320
> > [3390] 22/9/2022 -- 08:45:17 - (detect-engine-build.c:1784) <Info> (SigAddressCleanupStage1) -- cleaning up signature grouping structure... complete
> > [3390] 22/9/2022 -- 08:45:17 - (util-device.c:355) <Notice> (LiveDeviceListClean) -- Stats for '0000:3b:00.1':  pkts: 999999, drop: 0 (0.00%), invalid chksum: 0
> > [3390] 22/9/2022 -- 08:45:17 - (util-dpdk.c:52) <Info> (DPDKCloseDevice) -- Closing device 0000:3b:00.1
> > ```
> 
> That looks a bit... custom. Why the `___`s? Why the spaces/tabs between name and values?

It's a DPDK code snippet and I thought it looks fine. I have no strong feelings about that and can remove it.

**Comment by lukashino** [comment]
> > > > Can you add example output? Of the `PrintDPDKPortXstats` function?
> > > 
> > > 
> > > It prints out non-zero counters so there might be more or less output depending on how busy counters were. Here is an example:
> > > ```
> > > [3390] 22/9/2022 -- 08:44:34 - (tm-threads.c:1927) <Notice> (TmThreadWaitOnThreadInit) -- Threads created -> W: 1 FM: 1 FR: 1   Engine started.                                                                                                       
> > > [3390] 22/9/2022 -- 08:45:13 - (suricata.c:2776) <Notice> (SuricataMainLoop) -- Signal Received.  Stopping engine.
> > > [3390] 22/9/2022 -- 08:45:14 - (suricata.c:1168) <Info> (SCPrintElapsedTime) -- time elapsed 42.095s
> > > [3401] 22/9/2022 -- 08:45:15 - (flow-manager.c:1124) <Perf> (FlowRecycler) -- 41758 flows processed
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_packets:              999999
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_good_bytes:                760030582
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_packets:                999999
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_q0_bytes:          760030582
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_packets:           963516
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_unicast_bytes:             724508317
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_packets:         36483
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_multicast_bytes:           39522261
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_packets:               999999
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1): _______ rx_phy_bytes:         764030578
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:587) <Perf> (ReceiveDPDKThreadExitStats) -- Total RX stats of 0000:3b:00.1: packets 999999 bytes: 760030582 missed: 0 errors: 0 nombufs: 0
> > > [3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:598) <Perf> (ReceiveDPDKThreadExitStats) -- (W#01-0000..00.1) received packets 999999
> > > [3390] 22/9/2022 -- 08:45:16 - (counters.c:871) <Info> (StatsLogSummary) -- Alerts: 0
> > > [3390] 22/9/2022 -- 08:45:16 - (ippair.c:296) <Perf> (IPPairPrintStats) -- ippair memory usage: 414144 bytes, maximum: 16777216
> > > [3390] 22/9/2022 -- 08:45:17 - (host.c:299) <Perf> (HostPrintStats) -- host memory usage: 398144 bytes, maximum: 335544320
> > > [3390] 22/9/2022 -- 08:45:17 - (detect-engine-build.c:1784) <Info> (SigAddressCleanupStage1) -- cleaning up signature grouping structure... complete
> > > [3390] 22/9/2022 -- 08:45:17 - (util-device.c:355) <Notice> (LiveDeviceListClean) -- Stats for '0000:3b:00.1':  pkts: 999999, drop: 0 (0.00%), invalid chksum: 0
> > > [3390] 22/9/2022 -- 08:45:17 - (util-dpdk.c:52) <Info> (DPDKCloseDevice) -- Closing device 0000:3b:00.1
> > > ```
> > 
> > 
> > That looks a bit... custom. Why the `___`s? Why the spaces/tabs between name and values?
> 
> It's a DPDK code snippet and I thought it looks fine. I have no strong feelings about that and can remove it.

Changed output in the version 3 branch to:

```
[3401] 22/9/2022 -- 08:45:15 - (flow-manager.c:1124) <Perf> (FlowRecycler) -- 41758 flows processed
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_good_packets: 999999
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_good_bytes: 760030582
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_q0_packets: 999999
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_q0_bytes: 760030582
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_unicast_packets: 963516
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_unicast_bytes: 724508317
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_multicast_packets: 36483
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_multicast_bytes: 39522261
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_phy_packets: 999999
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:548) <Perf> (PrintDPDKPortXstats) -- Port 0 (0000:3b:00.1) - rx_phy_bytes: 764030578
[3399] 22/9/2022 -- 08:45:15 - (source-dpdk.c:587) <Perf> (ReceiveDPDKThreadExitStats) -- Total RX stats of 0000:3b:00.1: packets 999999 bytes: 760030582 missed: 0 errors: 0 nombufs: 0
```

**Pull request closed without merging**